crates/core_arch/src/nvptx/packed.rs

```diff
@@ -0,0 +1,107 @@
+//! NVPTX Packed data types (SIMD)
+//!
+//! Packed Data Types is what PTX calls SIMD types. See [PTX ISA (Packed Data Types)](https://docs.nvidia.com/cuda/parallel-thread-execution/#packed-data-types) for a full reference.
+//!
+//! Note: #[assert_instr] tests are not actually being run on nvptx due to being a `no_std` target incapable of running tests. Something like FileCheck would be appropriate for verifying the correct instruction is used.
+
+use crate::intrinsics::simd::*;
+
+#[allow(improper_ctypes)]
+extern "C" {
+    #[link_name = "llvm.minimum.v2f16"]
+    fn llvm_f16x2_min(a: f16x2, b: f16x2) -> f16x2;
+    #[link_name = "llvm.maximum.v2f16"]
+    fn llvm_f16x2_max(a: f16x2, b: f16x2) -> f16x2;
+}
+
+#[allow(unused)]
+macro_rules! simd_types {
+    ($(
+        $(#[$doc:meta])*
+        pub struct $name:ident($($fields:tt)*);
+    )*) => ($(
+        $(#[$doc])*
+        #[derive(Copy, Clone, Debug)]
+        #[allow(non_camel_case_types)]
+        #[repr(simd)]
+        #[allow(clippy::missing_inline_in_public_items)]
+        pub struct $name($($fields)*);
+    )*)
+}
+
+simd_types! {
+    /// PTX-specific 32-bit wide floating point (f16 x 2) vector type
+    #[unstable(feature = "stdarch_nvptx", issue = "111199")]
+    pub struct f16x2(f16, f16);
```

Comment:
You should use the `types!` macro to define SIMD types. Make sure to pull, the syntax for this macro just got changed.

Reply:
The syntax for the macro has changed, see the other architectures.